src/all_cats_photo.js

```diff
@@ -0,0 +1,74 @@
+(function () {
+    const selectElement = document.getElementById('friends');
+    const clickBtn = document.getElementById('clickBtn');
+    const popup = document.getElementById('popup');
+    const imagesContainer = document.getElementById('imagesContainer');
+
+    // Function to make API requests
+    async function getData(url) {
+        try {
+            const response = await fetch(url);
+            if (!response.ok) throw new Error(`Response status: ${response.status}`);
+            return await response.json();
+        } catch (error) {
+            console.error(error.message);
+            return error;
```

Comment:
If we want the function which will call this `getData` function to fail on error in `getData` it's better to throw here. Otherwise in `getBreedArray` you will see error related to absence of method `forEach`. And it's more confusing than the message which points to error in fetching data